torchaudio/__init__.py

```diff
@@ -274,6 +304,11 @@ def sox_encodinginfo_t():
         >>> ei.opposite_endian = torchaudio.get_sox_bool(0)
 
     """
+
+    if get_audio_backend() != "sox":
+        raise ImportError
```

Comment:
Instead of just raising an `ImportError`, it might be better user experience to explicitly say the "function YY not supported for backend XX".

Reply:
The trace should include the name of the function or you can maybe get the name of the function from locals()

Reply:
Agree, added along with decorator.